### DISCUSSION TITLE
Code examples for building applications with Next.js

### DISCUSSION THREAD

**Discussion by yanboishere** [uxuyconnect/connect-sdk#10] [General]
Some friends received an error about Next.js when building applications through the Solana documentation on the official website. Therefore, we provide SDK code examples.

**Comment by yanboishere** [comment]
The day before yesterday, Punky, a startup team from New York University, gave us feedback that after connecting the wallet according to the documentation, it can run in the browser environment, but an error is reported in the build phase.
![IMAGE 2024-10-30 03:14:55](https://github.com/user-attachments/assets/9bdda222-64e2-4caf-9523-41b1da29d06c)

**Comment by yanboishere** [comment]
![IMAGE 2024-10-30 03:20:18](https://github.com/user-attachments/assets/300839b4-9abf-4ab1-a008-93d8f1736068)

**Comment by yanboishere** [comment]
After that, our technique gives an example of Next.js：
https://github.com/uxuySafe/examples/blob/main/examples/next-pages/sdk/uxuyWallet.ts

      ’TypeScript‘

      import type { WalletTgSdk } from '@uxuycom/web3-tg-sdk';
      
      let SDk: WalletTgSdk | undefined;
      if (typeof window !== 'undefined') {
          const { WalletTgSdk } = require('@uxuycom/web3-tg-sdk');
          SDk = new WalletTgSdk();
      }
      
      
      export  function getSdk(){
          return SDk; 
      }   
      
      export function getTgEthereum(){
          return SDk?.ethereum;
      }
      
      export function getSolana(){
          return SDk?.solana;
      }
      
      
      
      export default {
          getSdk,
          getTgEthereum,
          getSolana
      };

**Comment by yanboishere** [comment]
If you have any related questions with SDK and other technology related issues in the future, please join our official developer and business Telegram community：

[UXUY Connect Official Group - How to used SDK](https://t.me/uxuyconnect/3)

or send email to ourselves:

bobby@uxuy.com

**Comment by Kali-Decoder** [comment]
what version are you using for integration of wallet in next js
